hwbench/engines/fio.py

```diff
@@ -0,0 +1,213 @@
+        return []
+
+    def parse_version(self, stdout: bytes, _stderr: bytes) -> bytes:
+        self.version = stdout.split(b"-")[1].strip()
+        return self.version
+
+    def version_major(self) -> int:
+        if self.version:
+            return int(self.version.split(b".")[0])
+        return 0
+
+    def version_minor(self) -> int:
+        if self.version:
+            return int(self.version.split(b".")[1])
+        return 0
+
+    def parse_cmd(self, stdout: bytes, stderr: bytes):
+        return {}
+
+
+class Fio(ExternalBench):
+    """The Fio stressor."""
+
+    def __init__(
+        self, engine_module: EngineModuleBase, parameters: BenchmarkParameters
+    ):
+        ExternalBench.__init__(self, engine_module, parameters)
+        self.parameters = parameters
+        self.engine_module = engine_module
+        self.log_avg_msec = 20000  # write_*_log are averaged at 20sec
+        self._parse_parameters()
+        # Tests can skip this part
+        if isinstance(parameters.out_dir, pathlib.PosixPath):
+            parameters.out_dir.joinpath("fio").mkdir(parents=True, exist_ok=True)
+
+    def version_compatible(self) -> bool:
+        engine = self.engine_module.get_engine()
+        return engine.version_major() >= 3 and engine.version_minor() >= 19
+
+    def _parse_parameters(self):
+        self.runtime = self.parameters.runtime
+        if self.runtime * 1000 < self.log_avg_msec:
+            fatal(
+                f"Fio runtime cannot be lower than the average log time ({self.log_avg_msec})."
+            )
+
+    def need_skip_because_version(self):
+        if self.skip:
+            # we already skipped this benchmark, we can't know the reason anymore
+            # because we might not have run the version command.
+            return ["echo", "skipped benchmark"]
+        if not self.version_compatible():
+            print(f"WARNING: skipping benchmark {self.name}, needs fio >= 3.19")
+            self.skip = True
+            return ["echo", "skipped benchmark"]
+        return None
+
+    def run_cmd(self) -> list[str]:
+        skip = self.need_skip_because_version()
+        if skip:
+            return skip
+
+        # Let's build the command line to run the tool
+        args = [
+            self.engine_module.get_engine().get_binary(),
+        ]
+
+        return self.get_taskset(args)
+
+    def get_default_fio_command_line(self, args: list) -> list:
+        """Return the default fio arguments"""
+
+        def remove_arg(args, item) -> list:
+            if isinstance(item, str):
+                return [arg for arg in args if not arg.startswith(item)]
+            else:
+                # We need to ensure that value based items are having the right value
+                # This avoid a case where the user already defined a value we need to control
+                for arg in args:
+                    if arg.startswith(item[0]):
+                        if arg != f"{item[0]}={item[1]}":
```

Comment:
The issue with this check is that it won't properly detect options like --numjobs 2, which are not separated by =